package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,459 @@
+            number of atoms in frame
+
+        Raises
+        ------
+        IOError
+        """
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'w':
+            raise IOError('File opened in mode: {}. Writing only allowed '
+                          'in mode "w"'.format('self.mode'))
+        if len(box) != 6:
+            raise ValueError("box size is wrong should be 6, got: {}".format(box.size))
+        if not self.wrote_header:
+            raise IOError("write header first before frames can be written")
+        xyz = np.asarray(xyz, order='F', dtype=np.float32)
+        if xyz.shape != (self.natoms, 3):
+            raise ValueError("xyz shape is wrong should be (natoms, 3), got:".format(xyz.shape))
+
+        cdef DOUBLE_T[::1] c_box = np.asarray(box, order='C', dtype=np.float64)
+        cdef FLOAT_T[::1] x = xyz[:, 0]
+        cdef FLOAT_T[::1] y = xyz[:, 1]
+        cdef FLOAT_T[::1] z = xyz[:, 2]
+
+        step = self.istart + self.current_frame * self.nsavc
+        ok = write_dcdstep(self.fp, self.current_frame + 1, step,
+                           self.natoms, <FLOAT_T*> &x[0],
+                           <FLOAT_T*> &y[0], <FLOAT_T*> &z[0],
+                           <DOUBLE_T*> &c_box[0], self.charmm)
+
+        self.current_frame += 1
+
+    def read(self):
+        if self.reached_eof:
+            raise IOError('Reached last frame in DCD, seek to 0')
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'r':
+            raise IOError('File opened in mode: {}. Reading only allow '
+                               'in mode "r"'.format('self.mode'))
+        if self.n_frames == 0:
+            raise IOError("opened empty file. No frames are saved")
+
+        cdef np.ndarray xyz = np.empty((self.natoms, self.ndims), dtype=FLOAT, order='F')
+        cdef np.ndarray unitcell = np.empty(6, dtype=DOUBLE)
+        unitcell[0] = unitcell[2] = unitcell[5] = 0.0;
+        unitcell[4] = unitcell[3] = unitcell[1] = 90.0;
+
+        cdef FLOAT_T[::1] x = xyz[:, 0]
+        cdef FLOAT_T[::1] y = xyz[:, 1]
+        cdef FLOAT_T[::1] z = xyz[:, 2]
+
+        first_frame = self.current_frame == 0
+        ok = read_dcdstep(self.fp, self.natoms,
+                          <FLOAT_T*> &x[0],
+                          <FLOAT_T*> &y[0], <FLOAT_T*> &z[0],
+                          <DOUBLE_T*> unitcell.data, self.nfixed, first_frame,
+                          self.freeind, self.fixedcoords,
+                          self.reverse_endian, self.charmm)
+        if ok != 0 and ok != -4:
+            raise IOError("Reading DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+        # we couldn't read any more frames.
+        if ok == -4:
+            self.reached_eof = True
+            raise StopIteration
+
+        self.current_frame += 1
+        return DCDFrame(xyz, unitcell)
+
+    def readframes(self, start=None, stop=None, step=None):
```

Comment:
Did we support subset reading before the libdcd rewrite? If not, I'd probably prefer to avoid making any more special features and focus on only supporting what old DCD did, but in Python 3 as well, then when the readers are excised to their own packages they can be perfected as needed.